presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRandomizeNullKeyInOuterJoin.java

```diff
@@ -119,6 +120,36 @@ public void testLeftJoinOnSameKey()
                 false);
     }
 
+    @Test
+    public void testLeftJoinOnSameKeyJoinAsRightSideInput()
+    {
+        assertPlan("select * from partsupp ps left join (select p.name, l.orderkey, l.partkey as partkey from part p left join lineitem l on p.partkey = l.partkey) pl on ps.partkey = pl.partkey",
+                Session.builder(getSessionAlwaysEnabled()).setSystemProperty(OPTIMIZE_HASH_GENERATION, "false").build(),
```

Comment:
To disable optimization hash generation, as this optimizer will recreate the join node and rearrange the join output, hence cover the problem here.